apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -406,7 +406,18 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
   public getShrinkwrapEntry(name: string, version: string): IPnpmShrinkwrapDependencyYaml | undefined {
     // Version can sometimes be in the form of a path that's already in the /name/version format.
     const packageId: string = version.indexOf('/') !== -1 ? version : `/${name}/${version}`;
-    return this._shrinkwrapJson.packages[packageId];
+    let ret: IPnpmShrinkwrapDependencyYaml | undefined = this._shrinkwrapJson.packages[packageId];
+    if (!ret) {
+      // Sometimes, the entry key could be /name/semver_peerDep@semver,
+      // So try to find the shrinkwrap dependency whose name starts with packageId_
+      for (const [k, v] of Object.entries(this._shrinkwrapJson.packages)) {
+        if (k.startsWith(`${packageId}_`)) {
+          ret = v;
+          break;
```

Comment:
What if there are multiple matches?  Should we print an error in that case?

Reply:
Is it possible to start with a package name/version_  with different suffix?

e.g.

`/foo/1.0.0_peerDep@1.0.0`

is it possible to have another `/foo/1.0.0_peerDep2@2.0.0`?

It seems impossible, but I could not 100% sure.

Reply:
Yes, this can definitely happen. You can reference the same package with multiple different peer dependency versions within the same repository. That's expected, in fact.

Reply:
Well, not within the same package's imports, but within the same repo, definitely.

Reply:
Any objections to merging this PR?  @D4N14L